qiskit_experiments/database_service/json.py

```diff
@@ -29,13 +32,25 @@ def default(self, obj: Any) -> Any:  # pylint: disable=arguments-differ
         if isinstance(obj, complex):
             return {"__type__": "complex", "__value__": [obj.real, obj.imag]}
         if isinstance(obj, FitVal):
-            return {"__type__": obj.__class__.__name__, "args": (obj.value, obj.stderr, obj.unit)}
-        if callable(obj):
-            return {"__type__": "callable", "__value__": obj.__name__}
+            return {"__type__": type(obj).__name__, "args": (obj.value, obj.stderr, obj.unit)}
```

Comment:
I would replace the check of `isinstance(obj, FitVal)` with:
```python
if dataclasses.is_dataclass(obj):
    return {"__type__": type(obj).__name__, "args": dataclasses.as_dict(obj)}
```

Reply:
Perhaps the `args` should be recursively passed through the `JSONEncoder` to ensure that you get valid JSON.

Reply:
Good idea, I wasn't sure how to instance check dataclasses since it didn't work with `isinstance`.

The way python encoder works that it does the recursive encoding/decoding already.